subsys/testsuite/ztest/include/zephyr/ztest_assert.h

```diff
@@ -141,6 +141,19 @@ static inline bool z_zassume(bool cond, const char *default_msg, const char *fil
 		}                                                                                  \
 	} while (0)
 
+/**
+ * @brief Skip the test, if @a cond is false
+ * It may be used to check if tested functionality supports expected features.
+ * E.g. in before function, you can check if charger supports assumed charging voltage and
+ * skip the test if it exceeds parameters used in test.
+ *
+ * You probably don't need to call this macro directly. You should
+ * instead use zassume_{condition} macros below.
+ *
+ * @param cond Condition to check
```

Comment:
This will conflict with https://github.com/zephyrproject-rtos/zephyr/pull/48667/files which I think will be merged today. So may want to hold off and rebase.

Reply:
Good to know, I will wait